sky/backends/cloud_vm_ray_backend.py

```diff
@@ -277,15 +282,17 @@ def check_ip():
                     for i in range(pg.bundle_count)
                 ])
                 print('INFO: Reserved IPs:', ip_list)
-                ip_list_str = '\\n'.join(ip_list)
-                sky_env_vars_dict['SKY_NODE_IPS'] = ip_list_str
 
                 if {cluster_ips_sorted!r}:
                     cluster_ips_map = {{ip: i for i, ip in enumerate({cluster_ips_sorted!r})}}
                     ip_rank_list = sorted(ip_list, key=cluster_ips_map.get)
                     ip_rank_map = {{ip: i for i, ip in enumerate(ip_rank_list)}}
+                    ip_list_str = '\\n'.join(ip_rank_list)
                 else:
                     ip_rank_map = {{ip: i for i, ip in enumerate(ip_list)}}
+                    ip_list_str = '\\n'.join(ip_list)
```

Comment:
nit: for readability, I would prefer to rename the original `ip_list` in L278 to `gang_scheduling_id_to_ip`.